components/gitpod-db/src/typeorm/migration/1695888330735-WorkspaceIndexBasedOnPrebuildId.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { indexExists } from "./helper/helper";
+
+const TABLE_NAME = "d_b_workspace";
+const INDEX_NAME = "ind_basedOnPrebuildId";
+
+export class WorkspaceIndexBasedOnPrebuildId1695888330735 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await indexExists(queryRunner, TABLE_NAME, INDEX_NAME))) {
+            await queryRunner.query(
+                `ALTER TABLE \`${TABLE_NAME}\` ADD INDEX \`${INDEX_NAME}\` (basedOnPrebuildId), ALGORITHM=INPLACE, LOCK=NONE`,
+            );
```

Comment:
how long do you expect this to take in GP Cloud? It used to be super expensive but Is this better with MySQL 8 now?

Reply:
It's not fundamentally faster, no.
This will take a while (couple of minutes), but I will execute it out-of-band once the PR is merged, so it does not clog the pipeline. :+1: